catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java

```diff
@@ -158,7 +158,11 @@ public MetacardImpl(Metacard metacard, MetacardType type) {
         }
         map = new HashMap<>();
         for (AttributeDescriptor attribute : metacard.getMetacardType().getAttributeDescriptors()) {
-            map.put(attribute.getName(), metacard.getAttribute(attribute.getName()));
+            Attribute metacardAttribute = metacard.getAttribute(attribute.getName());
```

Comment:
Should we add a note in the comment regarding the fact that `null` attributes and attribute values will _not_ be copied?